adr/0007-tenancy.md

```diff
@@ -0,0 +1,61 @@
+# Smallest Unit of Tenancy is the wasmCloud Host Struct
+
+This decision record covers the pros and cons and ultimate decision regarding multi-tenancy within
+a wasmCloud host.
+
+## Context and Problem Statement
+
+When running wasmCloud, people often need to run multiple copies of the same actor or capability provider. Running multiple copies of the capability provider with different link names gives us the ability to have multiple message brokers per actor, or multiple key-value stores, etc.
+
+When we run multiple copies of the same actor within the same host, we assume that these copies are _solely_ for horizontally scaling actors, and _not_ for multi-tenancy.
+
+An **inviolable** rule of wasmCloud's zero-trust policy is that we absolutely cannot ever allow one actor to compromise the security of another. We can't allow the generation of a link definition at runtime that could create a potential data exfiltration or PII compromise situation, or even enable the sharing of seemingly harmless data.
+
+Given the state of our documentation, people looking for multi-tenant solutions might not have sufficient context to make the right architectural decisions, and so this ADR needs to make clear where the tenancy boundaries are so that they can in turn be documented clearly at [wasmcloud.dev](https://wasmcloud.dev).
```

Comment:
wasmcloud.dev redirects to wasmcloud.com (might update URL if that's the new domain).

Reply:
Good catch. This is copied text from an old adr from when the .dev site was different.

Reply:
Wonderful to have these ADR's.. invaluable insight into the decision making. Another :+1: for wasmCloud :)

Reply:
Updated the links